crates/clvm-traits/src/match_byte.rs

```diff
@@ -5,19 +5,14 @@ use crate::{ClvmDecoder, ClvmEncoder, FromClvm, FromClvmError, ToClvm, ToClvmErr
 #[derive(Debug, Copy, Clone)]
 pub struct MatchByte<const BYTE: u8>;
 
-impl<N, const BYTE: u8> ToClvm<N> for MatchByte<BYTE> {
-    fn to_clvm(&self, encoder: &mut impl ClvmEncoder<Node = N>) -> Result<N, ToClvmError> {
-        if BYTE == 0 {
-            return encoder.encode_atom(&[]);
-        }
-        let number = BigInt::from(BYTE);
-        let bytes = number.to_signed_bytes_be();
-        encoder.encode_atom(&bytes)
+impl<N, E: ClvmEncoder<Node = N>, const BYTE: u8> ToClvm<E> for MatchByte<BYTE> {
+    fn to_clvm(&self, encoder: &mut E) -> Result<N, ToClvmError> {
+        encoder.encode_bigint(BigInt::from(BYTE))
```

Comment:
it seems like quite a detour to go via a `BitInt` here. In most cases, you'll most likely end up with the same byte as you started with, right?
Maybe it would make sense to do something like:
```
if BYTE == 0 { encoder.encode_atom(&[]) }
else if BYTE < 0x80 { encoder.encode_atom(&[BYTE]); } 
else { encoder.encode_atom(&[0, BYTE]); } 
```

Reply:
Changed it to this

Reply:
this is still very expensive compared to just `encode_atom()` with a single byte. Anything involving creating and destroying a `BigInt` is expensive. It's one of the main bottlenecks in `clvmr`